### PR TITLE
Implemented login and logout

### PR DESCRIPTION
**Pull request by rajzshkr**
Need clarification:

There is no argument for login and logout. So shall we remove `LoginOptions` and `toLoginOptions`?

### CODE REVIEW

File: src/utils/yarn.js [hunk 1]
```diff
@@ -155,3 +155,19 @@ export async function info(cwd: string, spawnArgs: Array<string> = []) {
     tty: true
   });
 }
+
+export async function login(cwd: string) {
```
Comment (ajaymathur) [inline src/utils/yarn.js]: can we reuse `cliCommands` function.

**Comment by ajaymathur** [comment]
We should keep using yarn for consistency. 🤔

**Comment by lukebatchelor** [comment]
We use npm for publishing already because of the issues we had with yarn

**Comment by sendilkumarn** [comment]
I do second that we should use `npm`

**Comment by rajzshkr** [comment]
Test case added!!

File: src/commands/__tests__/login.test.js [hunk 2]
```diff
@@ -1,4 +1,30 @@
...
-test('bolt login');
+describe('bolt login', () => {
+  let npmLoginSpy;
+
+  beforeEach(() => {
+    npmLoginSpy = jest.spyOn(npm, 'login');
+  });
+
+  afterEach(() => {
+    jest.clearAllMocks();
```
Comment (ajaymathur) [inline src/commands/__tests__/login.test.js]: no need to clear mocks.
just mock npm module like
`jest.mock('../../utils/npm')` // just after imports
we are already have config to reset mocks in jest.config.js in root

Comment (ajaymathur) [reply src/commands/__tests__/login.test.js]: sorry it is package json - https://github.com/boltpkg/bolt/blob/master/package.json#L76

Comment (rajzshkr) [reply src/commands/__tests__/login.test.js]: Updated!!

**Pull request merged**